### PR TITLE
add expression restorer

### PR DESCRIPTION
**Pull request by harisreedhar**

### CODE REVIEW

File: facefusion/processors/frame/face_restorer.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	prepare_source_frame = cv2.resize(source_vision_frame, (512, 512))[:, :, ::-1].transpose(2, 0, 1).astype(numpy.float32) / 255
+	prepare_target_frame = cv2.resize(target_vision_frame, (512, 512))[:, :, ::-1].transpose(2, 0, 1).astype(numpy.float32) / 255
+	prepare_restore_amount = numpy.array(restore_amount).astype(numpy.float32)
+
+	with conditional_thread_semaphore():
+		restore_frame : VisionFrame = expression_restorer.run(None,
+		{
+			'source': [ prepare_source_frame ],
+			'target': [ prepare_target_frame ],
+			'intensity': prepare_restore_amount
```
Comment (henryruhs) [inline facefusion/processors/frame/face_restorer.py:73]: rename inpu intensity to amount

File: facefusion/processors/frame/modules/face_swapper.py [hunk 2]
```diff
@@ -10,22 +11,26 @@
...
-from facefusion.common_helper import get_first
+from facefusion.common_helper import create_metavar, get_first
 from facefusion.content_analyser import clear_content_analyser
 from facefusion.download import conditional_download, is_download_done
 from facefusion.execution import create_inference_session, has_execution_provider
 from facefusion.face_analyser import clear_face_analyser, get_average_face, get_many_faces, get_one_face
 from facefusion.face_helper import paste_back, warp_face_by_face_landmark_5
-from facefusion.face_masker import clear_face_occluder, clear_face_parser, create_occlusion_mask, create_region_mask, create_static_box_mask
+from facefusion.face_masker import clear_face_occluder, clear_face_parser, create_occlusion_mask, create_region_mask, \
+	create_static_box_mask
```
Comment (henryruhs) [inline facefusion/processors/frame/modules/face_swapper.py:21]: fix import `, \`

File: facefusion/processors/frame/modules/face_swapper.py [hunk 3]
```diff
@@ -10,22 +11,26 @@
...
 from facefusion.face_selector import find_similar_faces, sort_and_filter_faces
 from facefusion.face_store import get_reference_faces
-from facefusion.filesystem import filter_image_paths, has_image, in_directory, is_file, is_image, is_video, resolve_relative_path, same_file_extension
+from facefusion.filesystem import filter_image_paths, has_image, in_directory, is_file, is_image, is_video, \
+	resolve_relative_path, same_file_extension
```
Comment (henryruhs) [inline facefusion/processors/frame/modules/face_swapper.py:25]: fix import `, \`

File: facefusion/processors/frame/modules/face_swapper.py [hunk 4]
```diff
@@ -10,22 +11,26 @@
...
 from facefusion.processors.frame import choices as frame_processors_choices
+from facefusion.processors.frame.face_restorer import clear_expression_restorer, restore_expression
 from facefusion.processors.frame.pixel_boost import explode_pixel_boost, implode_pixel_boost
 from facefusion.processors.frame.typing import FaceSwapperInputs
 from facefusion.program_helper import find_argument_group, suggest_face_swapper_pixel_boost_choices
 from facefusion.thread_helper import conditional_thread_semaphore, thread_lock
-from facefusion.typing import Args, Embedding, Face, ModelSet, OptionsWithModel, ProcessMode, QueuePayload, UpdateProgress, VisionFrame
+from facefusion.typing import Args, Embedding, Face, ModelSet, OptionsWithModel, ProcessMode, QueuePayload, \
+	UpdateProgress, VisionFrame
```
Comment (henryruhs) [inline facefusion/processors/frame/modules/face_swapper.py:33]: fix import `, \`

File: facefusion/processors/frame/modules/face_swapper.py [hunk 5]
```diff
@@ -264,6 +271,7 @@ def post_process() -> None:
 		clear_content_analyser()
 		clear_face_occluder()
 		clear_face_parser()
+		clear_expression_restorer()
```
Comment (henryruhs) [inline facefusion/processors/frame/modules/face_swapper.py:274]: file name is face restorer, here you name it expression restorer...

File: facefusion/processors/frame/modules/face_swapper.py [hunk 6]
```diff
@@ -292,6 +301,10 @@ def swap_face(source_face : Face, target_face : Face, temp_vision_frame : Vision
 		region_mask = create_region_mask(crop_vision_frame, state_manager.get_item('face_mask_regions'))
 		crop_masks.append(region_mask)
 	crop_mask = numpy.minimum.reduce(crop_masks).clip(0, 1)
+	if state_manager.get_item('face_swapper_expression_restorer') != 0:
```
Comment (henryruhs) [inline facefusion/processors/frame/modules/face_swapper.py:304]: use > 0

File: facefusion/uis/components/frame_processors_options.py [hunk 7]
```diff
@@ -156,18 +168,18 @@ def listen() -> None:
...
 	if frame_processors_checkbox_group:
-		frame_processors_checkbox_group.change(update_frame_processors, inputs = frame_processors_checkbox_group, outputs = [ AGE_MODIFIER_MODEL_DROPDOWN, AGE_MODIFIER_DIRECTION_SLIDER, FACE_DEBUGGER_ITEMS_CHECKBOX_GROUP, FACE_ENHANCER_MODEL_DROPDOWN, FACE_ENHANCER_BLEND_SLIDER, FACE_SWAPPER_MODEL_DROPDOWN, FACE_SWAPPER_PIXEL_BOOST_DROPDOWN, FRAME_COLORIZER_MODEL_DROPDOWN, FRAME_COLORIZER_BLEND_SLIDER, FRAME_COLORIZER_SIZE_DROPDOWN, FRAME_ENHANCER_MODEL_DROPDOWN, FRAME_ENHANCER_BLEND_SLIDER, LIP_SYNCER_MODEL_DROPDOWN ])
+		frame_processors_checkbox_group.change(update_frame_processors, inputs = frame_processors_checkbox_group, outputs = [ AGE_MODIFIER_MODEL_DROPDOWN, AGE_MODIFIER_DIRECTION_SLIDER, FACE_DEBUGGER_ITEMS_CHECKBOX_GROUP, FACE_ENHANCER_MODEL_DROPDOWN, FACE_ENHANCER_BLEND_SLIDER, FACE_SWAPPER_MODEL_DROPDOWN, FACE_SWAPPER_PIXEL_BOOST_DROPDOWN, FACE_SWAPPER_EXPRESSION_RESTORER_SLIDER, FRAME_COLORIZER_MODEL_DROPDOWN, FRAME_COLORIZER_BLEND_SLIDER, FRAME_COLORIZER_SIZE_DROPDOWN, FRAME_ENHANCER_MODEL_DROPDOWN, FRAME_ENHANCER_BLEND_SLIDER, LIP_SYNCER_MODEL_DROPDOWN ])
```
Comment (henryruhs) [inline facefusion/uis/components/frame_processors_options.py:171]: this is getting unmaintainable... ideas?

File: facefusion/processors/frame/face_restorer.py [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+from facefusion.execution import create_inference_session
+from facefusion.face_masker import create_face_mask
+from facefusion.filesystem import is_file, resolve_relative_path
+from facefusion.thread_helper import conditional_thread_semaphore, thread_lock
+from facefusion.typing import ModelSet, VisionFrame
+
+EXPRESSION_RESTORER = None
+MODELS : ModelSet =\
+{
+	'live_portrait_expression_restorer':
```
Comment (henryruhs) [inline facefusion/processors/frame/face_restorer.py:18]: just 'expression_restorer'

File: facefusion/face_masker.py [hunk 9]
```diff
@@ -159,3 +159,8 @@ def create_mouth_mask(face_landmark_68 : FaceLandmark68) -> Mask:
 	mouth_mask = cv2.erode(mouth_mask.clip(0, 1), numpy.ones((21, 3)))
 	mouth_mask = cv2.GaussianBlur(mouth_mask, (0, 0), sigmaX = 1, sigmaY = 15)
 	return mouth_mask
+
+
+def create_face_mask(crop_vision_frame : VisionFrame) -> Mask:
```
Comment (henryruhs) [inline facefusion/face_masker.py:164]: I wonder why this was not possible for the mouth mask :-)

Comment (harisreedhar) [reply facefusion/face_masker.py:164]: doing with landmarker is cheaper

**Pull request merged**